File: Java/Reverse_String.java

public class Reverse_String {
	
    public void reverseString(char[] s) {
        for (int i = 0; i < s.length / 2; i++) {
        	char temp = s[i];
        	int last = s.length - 1 - i;
        	s[i] = s[last];
        	s[last] = temp;
        }
    }

}

File: Typescript/Reverse_String.ts
function reverseString(s: string[]): void {
	let endPointer: number = s.length - 1;
	for (let i = 0; i < s.length / 2; i++) {
		let temp: string = s[i];
		s[i] = s[endPointer];
		s[endPointer] = temp;
		endPointer--;
	}
}
